### PR TITLE
Fixes exploded circuit shells attempting to remove a deleted circuit

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About The Pull Request
Fixes a runtime that would occur when a circuit shell was inflicted with a devastating explosion and the shell attempted to remove the circuit after it had been deleted.

## Why It's Good For The Game
One less runtime.

### CODE REVIEW

File: code/datums/components/shell.dm [hunk 1]
```diff
@@ -106,8 +106,13 @@
 
 /datum/component/shell/proc/on_object_deconstruct()
 	SIGNAL_HANDLER
-	if(!(shell_flags & SHELL_FLAG_CIRCUIT_FIXED) && !attached_circuit?.admin_only)
-		remove_circuit()
+	if(!attached_circuit)
+		return
+	if( attached_circuit.admin_only)
```
Comment (Watermelon914) [inline code/datums/components/shell.dm:111]: remove the extra spaces here

File: code/datums/components/shell.dm [hunk 2]
```diff
@@ -106,8 +106,13 @@
...
+		return
+	if( shell_flags & SHELL_FLAG_CIRCUIT_FIXED)
```
Comment (Watermelon914) [inline code/datums/components/shell.dm:113]: same here

File: code/datums/components/shell.dm (hunk 1 above)
Comment (TemporalOroboros) [reply code/datums/components/shell.dm]: Done.

File: code/datums/components/shell.dm (hunk 2 above)
Comment (TemporalOroboros) [reply code/datums/components/shell.dm]: Done.

**Comment by LemonInTheDark** [comment]
I take it not having an attached circuit is an acceptable state at this point?

**Comment by Watermelon914** [comment]
> I take it not having an attached circuit is an acceptable state at this point?

yes, shells don't need an attached circuit.
Some signals that are handled rely on an attached circuit existing 100% of the time though

**Pull request merged**